### PR TITLE
Clarifications of stored-definition

### PR DESCRIPTION
**Pull request by henrikt-ma**
I got a question today where I believe the clarifications we discussed earlier in #3466 would have helped, so I thought it was time to turn them into a PR.

### CODE REVIEW

File: chapters/packages.tex [hunk 1]
```diff
@@ -227,12 +227,9 @@ \subsection{Directory Hierarchy Mapping}\label{mapping-a-package-class-hierarchy
...
 \subsection{Single File Mapping}\label{mapping-a-package-class-hierarchy-into-a-single-file-nonstructured-entity}\label{single-file-mapping}
 
-When mapping a package or class-hierarchy to a file (e.g., the file \filename{A.mo}), that file shall only define a single class \lstinline!A! with a name matching the name of the nonstructured entity.
-In a file hierarchy the files shall have the extension \filename{.mo}.
-
-A \filename{.mo} file defining more than one class cannot be part of the mapping
-to file-structure and it is an error if it is loaded from the
-\lstinline!MODELICAPATH!.
+When mapping a package or class-hierarchy to a file (e.g., the file \filename{A.mo}), the file content shall match \lstinline[language=grammar]!stored-definition! in the grammar.
+In this case, the \lstinline[language=grammar]!stored-definition! shall only define a single class with name (here, \lstinline!A!) matching the name of the nonstructured entity.
```
Comment (HansOlsson) [inline chapters/packages.tex:231]:
```suggestion
In this case, the \lstinline[language=grammar]!stored-definition! shall only define a single class with a name (here, \lstinline!A!) matching the name of the nonstructured entity.
```
I think the indefinite article should be used here.

**Review by HansOlsson — APPROVED** [review]
Looks good except one minor issue.
It also made me realize another issue - will open separate PR

File: chapters/packages.tex (hunk 1 above)
Comment (HansOlsson) [reply chapters/packages.tex:231]: @henrikt-ma do you agree with the suggested change?

File: chapters/packages.tex [hunk 2]
```diff
@@ -227,12 +227,9 @@ \subsection{Directory Hierarchy Mapping}\label{mapping-a-package-class-hierarchy
 
 \subsection{Single File Mapping}\label{mapping-a-package-class-hierarchy-into-a-single-file-nonstructured-entity}\label{single-file-mapping}
 
-When mapping a package or class-hierarchy to a file (e.g., the file \filename{A.mo}), that file shall only define a single class \lstinline!A! with a name matching the name of the nonstructured entity.
-In a file hierarchy the files shall have the extension \filename{.mo}.
-
-A \filename{.mo} file defining more than one class cannot be part of the mapping
-to file-structure and it is an error if it is loaded from the
-\lstinline!MODELICAPATH!.
+When mapping a package or class-hierarchy to a file (e.g., the file \filename{A.mo}), the file content shall match \lstinline[language=grammar]!stored-definition! in the grammar.
```
Comment (eshmoylova) [inline chapters/packages.tex:230]: What is class-hierarchy if it is not a package? And also is it class-hierarchy or class hierarchy?

File: chapters/packages.tex (hunk 1 above)
Comment (henrikt-ma) [reply chapters/packages.tex:231]: No, my non-English ears don't like the presence of _a_ here.  When _a_ is there, I get the impression that the class could have any number names, but without _a_ I get the impression that it is a given that every class has a name.

File: chapters/packages.tex (hunk 2 above)
Comment (henrikt-ma) [reply chapters/packages.tex:230]: This is a class hierarchy which not a package:
```
model A
  record R
  end R;
  R r;
end A;
```

This is the only occurrence of hyphened _class-hierarchy_, so it should clearly be spaced.  I'll fix.

File: chapters/packages.tex (hunk 1 above)
Comment (HansOlsson) [inline chapters/packages.tex:231]: So, what about:
```suggestion
In this case, the \lstinline[language=grammar]!stored-definition! shall only define a single class whose name (here, \lstinline!A!) matches the name of the nonstructured entity.
```

Comment (henrikt-ma) [reply chapters/packages.tex:231]: Yes, that sounds better to me.  Applying.

Comment (henrikt-ma) [reply chapters/packages.tex]: This was settled in a later conversation.  Resolving.

Comment (henrikt-ma) [reply chapters/packages.tex]: By the way, @eshmoylova, did this sound like an improvement of language to you?

**Pull request merged**